doc/nrf/ug_fw_update.rst

```diff
@@ -0,0 +1,311 @@
+   If a build directory is removed and rebuilt from scratch, relying on the default ECDSA key for signing an application or upgradable second-stage bootloader image will result in a failed boot chain validation, if you are not programming also the |NSIB|.
+
+   You can avoid this issue by storing a custom private key outside of the build directory during development.
+
+.. _ug_fw_update_key_revocation:
+
+Public key revocation
+=====================
+
+The |NSIB| supports the revocation of public signing keys that are available to validate the next image in the secure boot chain.
+Key revocation can be an useful security or backup tool for devices that have already been deployed to the field.
+If a private key has been compromised or lost, its public key can be invalidated by uploading a new firmware image signed by another key known to the bootloader.
+
+These keys are kept internally by the bootloader, so the list of available public keys cannot change once it is deployed.
+See :option:`CONFIG_SB_PUBLIC_KEY_FILES` for details on how this mechanism is implemented.
+
+You can add this feature to your own project and check its functionality as follows:
+
+1. Use this :ref:`Python tool <ug_fw_update_keys_python>` to generate two or more private keys for the application and then extract a public key for each one:
+
+   .. code-block:: console
+
+      python3 nrf/scripts/bootloader/keygen.py --public -i priv.pem -o pub.pem
+
+#. Compile the application and bootloader with the relevant configurations, using only absolute paths:
+
+   .. code-block:: console
+
+      CONFIG_SECURE_BOOT=y
+      CONFIG_SB_SIGNING_KEY_FILE="/path/to/priv_a.pem"
+      CONFIG_SB_PUBLIC_KEY_FILES="/path/to/pub_b.pem,/path/to/pub_c.pem"
+
+   .. note::
+
+      The public key associated with the original private signing key should not be included in the public key list.
+
+#. Program the application to the target development kit and :ref:`check its console output <ug_bootloader_adding_testing>`.
+   With the first firmware version, ``priv_a.pem`` and ``pub_a.pem`` are used for signing and validating the image.
+
+   .. code-block:: console
+
+      *** Booting Zephyr OS build ...  ***
+      Attempting to boot slot 0.
+      Attempting to boot from address 0x9000.
+      Verifying signature against key 0.
+      Hash: 0xda...4f
+      Firmware signature verified.
+      Firmware version 1
+      *** Booting Zephyr OS build ...  ***
+      ...
+
+#. To revoke a single or multiple keys, rebuild the application using the private key associated with a key listed *after* the currently used key in the list.
+   For example, when compiling with the ``priv_c.pem`` key, images signed with ``priv_a.pem`` or ``priv_b.pem`` will no longer boot when uploaded into an image slot.
+
+   Modify the configuration settings as follows, including a new firmware version:
```

Comment:
Looks like this step includes two steps: rebuild and modify. Maybe split it into substeps?

Reply:
fixed